.github/workflows/ci-comment-pr-binaries.yml

```diff
@@ -0,0 +1,89 @@
+name: PR Comment with PR Binaries
```

Comment:
Can we just document this in either the README or CONTRIBUTING.md?  It seems relatively low value for day to day work (most people that actually want to do this will know how to), and it's easy to see if binaries are built correctly by just looking at the CI status.  Adding an additional comment to each PR just makes the PR page a lot noisier.

Reply:
I that it's likely README/CONTRIBUTING is not self-documenting enough.

Eventually I would like this to work for end users as well, but it wasn't something I got to in my hackathon. That would entail publishing to S3 and updating the get.pulumi.com script - but then we could direct a user to this comment to try out a fix. That seems quite high value to me.

Reply:
End users barely use dev releases, which are easy to install and use even in CI, so I kind of doubt they would use this a lot.  It does seem like a useful functionality when we try to get end-users to test a PR, but we usually have to direct them to do that anyway in an issue comment, and it's easy enough for us to pull that comment together.  I'm fairly strongly against adding this additional comment.  I could be convinced to add a line to the changelog comment as I said, but a separate comment just makes PRs harder to read, which I really want to avoid.

Reply:
I have to push back for a couple reasons: 

1. The `on-pr-changelog` job runs on both contributor (branch) and community (fork) pull requests. The latter don't trigger builds, and won't produce consumable artifacts.

2. The primitives we have access to - find comment, find and update comment - don't compose in a way that allows us to "decouple" these comments. It's not trivial to "just edit" the relevant portion of an existing comment.

While I appreciate striving for "simple" PR pages, many people have arrived at similar conclusions. Take a look at a [Kubernetes PR](https://github.com/kubernetes/kubernetes/pull/126854) and there are no fewer than four comments from bots. I'd argue we should let each "bot" or CI automation do one thing it's good at, rather than try to combine these when the primitives we have (GitHub Actions) are not as composable as a homegrown automation. Comments are the tool we have, and we already take care to make sure those comments are created once and updated to be correct as a PR is updated. We're setting a high bar for ourselves already.

Reply:
@AaronFriel we discussed this yesterday in the team meeting and decided as a team that we'd rather not have this additional comment in each PR.  Some of the reasons:
- This clutters the PR page.  There's already a changelog comment that pretty much everyone learned to ignore, and we'd rather not have more comments that we need to ignore, as that just wastes time.
- We don't think it's actually a good idea for end users to install binaries from PRs without us prompting them to do that.  The code we push into PRs is often still untested for collaboration purposes, and at the very least a comment like this needs a much bigger warning to be very careful with it.  Meanwhile Pulumi employees that need this frequently can get this out of the documentation.
- This doesn't work for windows users, so it makes Windows feel more like a second class platform, which we don't want.
- (there may have been more that I'm forgetting about now)

We discussed that having a single comment with all collapsed sections in the 

<details>
<summary>like this</summary>

</details>

might be acceptable, but we would like to avoid the proliferation of comments (with some having stronger and others less strong opinions).

Another option we had discussed (not sure how easy this is), would be to have a link on a PR check that links to a page with instructions.  Similar to what @lunaris added with the docs check.